static/js/stream_edit.js

```diff
@@ -353,12 +353,6 @@ function stream_setting_clicked(e) {
         }
     }
     exports.set_stream_property(sub, setting, !sub[setting]);
-
-    // A hack.  Don't change the state of the checkbox if we
-    // clicked on the checkbox itself.
-    if (checkbox[0] !== e.target) {
-        checkbox.prop("checked", !checkbox.prop("checked"));
-    }
```

Comment:
I think this does make for odd behavior if the we add latency (a `time.sleep(5)` in `update_subscription_properties_backend`); one ends up likely to click the checkbox several times to see it change.

I think probably this commit is still an improvement; but we should prioritize adding the same "Saving.." indicator that we use in other user settings for this widget.

(Which also should be converted to use the same styling as our other user-level settings).

Reply:
Sure, ill add this as a followup as we're reusing the function from `stream_edit`, there should be a saving indicator for the stream specific notifications (to match with general notifications) as well as in the stream edit page.